From d2d74d6aaa43af3dd1735d5afdc51ccd2f78e837 Mon Sep 17 00:00:00 2001
Message-Id: <d2d74d6aaa43af3dd1735d5afdc51ccd2f78e837.1525290244.git.christoph@terra.fritz.box>
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Fri, 24 Nov 2017 11:59:50 +0100
Subject: [PATCH 1/3] Improve setup of the schema cache location

1. Prevent from errors when the host is configured e.g. with ldap:// prefix
2. Use sys_get_temp_dir() to get the system tmp directory
---
 lib/Net/LDAP3.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/Net/LDAP3.php b/lib/Net/LDAP3.php
index 8872e47..b5dcae3 100644
--- a/lib/Net/LDAP3.php
+++ b/lib/Net/LDAP3.php
@@ -2149,8 +2149,9 @@ class Net_LDAP3
             );
 
             $_ldap_schema_cache_cfg = array(
-                'path' => "/tmp/" . $host . ":" . ($port ? $port : '389') . "-Net_LDAP2_Schema.cache",
                 'max_age' => 86400,
+                'path' => sprintf('%s/%s:%d-Net_LDAP2_Schema.cache',
+                    (sys_get_temp_dir() ?: '/tmp'), str_replace('://', ':', $host) . (strpos($host, ":$port") ? '' : ":$port")),
             );
 
             $_ldap = Net_LDAP2::connect($_ldap_cfg);
-- 
2.14.1

